MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -3336,46 +3361,20 @@ InitializePageAttributesForMemoryProtectionPolicy (
     ASSERT (StackBase != 0);
   }
 
-  if (EFI_ERROR (Status)) {
-    return;
-  }
-
-  DEBUG ((
-    DEBUG_INFO,
-    "%a: applying strict permissions to active memory regions\n",
-    __FUNCTION__
-    ));
-
-  MemoryMapEntry = MemoryMap;
-  MemoryMapEnd   = (EFI_MEMORY_DESCRIPTOR *)((UINT8 *)MemoryMap + MemoryMapSize);
-  while ((UINTN)MemoryMapEntry < (UINTN)MemoryMapEnd) {
-    if (MemoryMapEntry->Attribute != 0) {
-      SetUefiImageMemoryAttributes (
-        MemoryMapEntry->PhysicalStart,
-        LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT),
-        MemoryMapEntry->Attribute
-        );
-
-      // Add EFI_MEMORY_RP attribute for the first page of the stack if stack
-      // guard is enabled.
-      if ((StackBase != 0) &&
-          ((StackBase >= MemoryMapEntry->PhysicalStart) &&
-           (StackBase <  MemoryMapEntry->PhysicalStart +
-            LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT))) &&
-          gDxeMps.CpuStackGuard)
-      {
-        SetUefiImageMemoryAttributes (
-          StackBase,
-          EFI_PAGES_TO_SIZE (1),
-          EFI_MEMORY_RP | MemoryMapEntry->Attribute
-          );
-      }
-    }
-
-    MemoryMapEntry = NEXT_MEMORY_DESCRIPTOR (MemoryMapEntry, DescriptorSize);
+  // Add EFI_MEMORY_RP attribute for the first page of the stack if stack
+  // guard is enabled.
+  if ((StackBase != 0) &&
+      ((StackBase >= MemoryMapEntry->PhysicalStart) &&
+       (StackBase <  MemoryMapEntry->PhysicalStart +
+        LShiftU64 (MemoryMapEntry->NumberOfPages, EFI_PAGE_SHIFT))) &&

```

Comment:
This check needs to be updated, right, because we are no longer checking per entry and this would be the final memory map entry?

Reply:
Yes, good catch 👍